server/pulp/server/db/model/__init__.py

```diff
@@ -307,3 +312,132 @@ def post_save(cls, sender, document, **kwargs):
 
 
 signals.post_save.connect(TaskStatus.post_save, sender=TaskStatus)
+
+
+class ContentUnit(Document):
+    """
+    The base class for all content units.
+
+    All unit inheriting from this class should override the unit_type_id and _ns to ensure
+    that they are populated properly
+
+
+    :ivar id: string representation of content unit id
+    :type id: mongoengine.StringField
+    :ivar last_updated: integer representation of last time this unit was updated (mer
+    :type last_updated: mongoengine.IntField
+    :ivar user_metadata: Bag of user supplied data to go along with this unit
+    :type user_metadata: mongoengine.DictField
+    :ivar storage_path: Location on disk where the content associated with this unit lives
+    :type storage_path: mongoengine.StringField
+
+    :ivar _ns: The namespace field (Deprecated), reading
+    :type _ns: mongoengine.StringField
+    :ivar unit_type_id: string representation of content unit type
+    :type unit_type_id: mongoengine.StringField
+    """
+
+    id = StringField(primary_key=True)
```

Comment:
Will this cause problems like the ones we were discussing last week where the 'id' attribute is aliased to _id internally in mongoengine?